raft.go

```diff
@@ -927,6 +940,23 @@ func (r *Raft) appendEntries(rpc RPC, a *AppendEntriesRequest) {
 	return
 }
 
+// checkAndProcessConfigurationLog takes a log entry and updates the latest
+// configuration if the entry results in a new configuration. This must only be
+// called from the main thread.
```

Comment:
I don't think we want to relax this restriction with any locks since this should only apply in the main thread, ever. I'll update the comment about getting called from `NewRaft()` that's before any threads are running.